src/vorta/views/extract_dialog.py

```diff
@@ -532,7 +532,7 @@ def data(self, index: QModelIndex, role: int = Qt.ItemDataRole.DisplayRole):
         if role == Qt.ItemDataRole.CheckStateRole and column == 0:
             return item.data.checkstate
 
-    def setData(self, index: QModelIndex, value, role: int = Qt.ItemDataRole.DisplayRole) -> bool:
+    def setData(self, index: QModelIndex, value: Qt.CheckState, role: int = Qt.ItemDataRole.DisplayRole) -> bool:
```

Comment:
Right. Anything that could be improved? Should we have a hard check to validate the type? Like for role.

Reply:
I don't think that's needed. We don't do type checks in any other python code. However the type hints should be updated. While value can remain untyped in the signature (it could be cast to `CheckState` later though), role might not be of type `int` any more. Someone has to check that and any other methods expecting a qt enum.

Reply:
> role might not be of type `int` any more. Someone has to check that and any other methods expecting a qt enum.

The Qt6 docs states `int role = Qt::DisplayRole)` it still is of type int in PyQt6.

Reply:
The comparison works with ints, yes. I do think we mostly get `role` as type `Qt.ItemDataRole.DisplayRole` and then compare it to be the same. So shouldn't need any change.

```
In [7]: Qt.ItemDataRole.DisplayRole == 0
Out[7]: True
```

Reply:
I do wonder if this would be a better type hint?

`def data(self, index: QModelIndex, role: Qt.ItemDataRole = Qt.ItemDataRole.DisplayRole):`

Reply:
We use roles mostly as `DisplayRole`, then I think it makes sense.

Reply:
```suggestion
    def setData(self, index: QModelIndex, value, role: Union[int, Qt.ItemDataRole] = Qt.ItemDataRole.DisplayRole) -> bool:
```

will be best since Qt passes `int` for `role` while devs (we) might want to pass `ItemDataRole`.

Reply:
> will be best since Qt passes `int` for `role` while devs (we) might want to pass `ItemDataRole`.

Ya you're right, that would be more helpful as a type hint.

Reply:
Pushed

Reply:
The Or for type hints was introduced with py3.10 though.

Reply:
Oops. Shows up in the tests.